runtime/gc_base/StringTable.cpp

```diff
@@ -634,17 +665,39 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 				}
 			}
 		} else {
-			if (isCompressable) {
+			if (IS_STRING_COMPRESSION_ENABLED_VM(vm) && isCompressible) {
 				vm->internalVMFunctions->copyUTF8ToCompressedUnicode(vmThread, data, length, stringFlags, charArray, 0);
 			} else {
-				vm->internalVMFunctions->copyUTF8ToUnicode(vmThread, data, length, stringFlags, charArray, 0);
+				if (isCompressible) {
+					if (J9_ARE_ANY_BITS_SET(stringFlags, J9_STR_XLAT)) {
+						for (UDATA i = 0; i < length; ++i) {
+							U_8 c = data[i];
+							J9JAVAARRAYOFCHAR_STORE(vmThread, charArray, i, c != '/' ? c : '.');
+						}
+					} else {
+						for (UDATA i = 0; i < length; ++i) {
+							J9JAVAARRAYOFCHAR_STORE(vmThread, charArray, i, data[i]);
+						}
+					}
+
+					if (J9_ARE_ALL_BITS_SET(stringFlags, J9_STR_ANON_CLASS_NAME)) {
+						for (IDATA i = length - 1; i >= 0; --i) {
+							if ('.' == J9JAVAARRAYOFCHAR_LOAD(vmThread, charArray, i)) {
+								J9JAVAARRAYOFCHAR_STORE(vmThread, charArray, i, '/');
+								break;
+							}
+						}
+					}
```

Comment:
I would prefer to see inline versions of these functions developed rather than duplicating the logic.

Reply:
Fixed in 2cb4393 and 3ba7fc7